docs/ref/migration-operations.txt

```diff
@@ -402,7 +402,18 @@ You can take this template and work from it, though we suggest looking at the
 built-in Django operations in ``django.db.migrations.operations`` - they're
 easy to read and cover a lot of the example usage of semi-internal aspects
 of the migration framework like ``ProjectState`` and the patterns used to get
-historical models.
+historical models, as well as ``ModelState`` and the patterns used to mutate
+historical models in ``state_forwards``.
+
+.. warning::
```

Comment:
Do you think this is a common issue? I am not sure a warning box is needed. The placing seems a bit odd (before what seem to be more basic instructions).